docs/proposals/service-design.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Service Design
+status: Pending
+authors:
+    - "@qizha"
+approvers:
+  - "@kevin-wangzefeng"
+  - "@CindyXing"
+creation-date: 2019-03-20
+last-updated: 2019-03-20
+---
+
+# Service support on edge
+
+## Abstract
+To support service capabilities on edge to support microservice communication cross cloud and edge.
+The service discovery, commuunication should be considered in this feature.
+Developers can define services on top of the ddeployments deployed on the cloud or edges in a unified manner without caring about their location. The service name is the only thing they need to know to do the communication. For example, the following address can be used for a REST  request:
+`http://{service_name}/resources`
+
+## Motivation
+The cloud native and microservice architecture becoming more and more popular and the edge node is becoming more and more powerful. The user want to separate their application to multiple microservice and some of them need to be deployed in edge node to get close to the data generated by deivces. By simply the development, the user need to use the same michanism to do the service discovery and communication with what they are doing on cloud.
+
+## Constraints and Assumptions
+1. Since the edge environment can be some non-standard system to running on some specific devices so we should not rely on the kernel network capability. In that way, we need to use userspace proxy to implement the data proxy on edge.
```

Comment:
some non-standard system to running ~~on~~ some specific devices,  ~~so~~ we should not rely on the kernel network capability.